ui-common/src/components/TrafficViewer/Filters.tsx

```diff
@@ -47,18 +54,59 @@ export const modalStyle = {
     color: '#000',
 };
 
-export const QueryForm: React.FC<QueryFormProps> = ({backgroundColor, reopenConnection}) => {
+export const CodeEditorWrap: FC<QueryFormProps> = ({ query, onQueryChange, validateQuery, onValidationChanged }) => {
+    const [queryBackgroundColor, setQueryBackgroundColor] = useState("#f5f5f5");
+    const handleQueryChange = useMemo(
+        () =>
+            debounce(async (query: string) => {
+                if (!query) {
+                    setQueryBackgroundColor("#f5f5f5");
+                    onValidationChanged && onValidationChanged({ query: query, message: "", valid: true })
```

Comment:
why onCalidationChanged Twice?

Reply:
you mean called twice?
what would you suggest?

Reply:
I missed the "?" in interface. I got it now. Is there a case where you don't want to use "onValidationChanged" in CodeEditorWrap?